From 33b99146840898cf20d89af746f81e3eaee5d7fc Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Tue, 19 Nov 2024 17:09:24 +0100
Subject: [PATCH] dt-bindings: iio: stm32-mdf: document channels binding

The IIO channels were initially documented in the MDF binding specific
to the analog part. The audio nodes also use an IIO channel.
Update the MDF binding to document channels for both audio and analog
parts.

Change-Id: Ic430297ae854829dbebc526a27756adee4b24c2b
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/420090
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 .../bindings/iio/adc/st,stm32-mdf-adc.yaml    | 75 +++++++++----------
 1 file changed, 36 insertions(+), 39 deletions(-)

--- a/Documentation/devicetree/bindings/iio/adc/st,stm32-mdf-adc.yaml
+++ b/Documentation/devicetree/bindings/iio/adc/st,stm32-mdf-adc.yaml
@@ -164,6 +164,12 @@ patternProperties:
       "#io-channel-cells":
         const: 1
 
+      '#address-cells':
+        const: 1
+
+      '#size-cells':
+        const: 0
+
       st,cic-mode:
         description: |
           Cascaded-integrator-comb (CIC) filter configuration
@@ -215,8 +221,31 @@ patternProperties:
       - reg
       - interrupts
       - "#io-channel-cells"
+      - "#address-cells"
+      - "#size-cells"
       - st,sitf
 
+    patternProperties:
+      "^channel@([0-7])$":
+        type: object
+        $ref: "adc.yaml"
+        description: Represents the external channel which is connected to the MDF.
+
+        properties:
+          reg:
+            maximum: 7
+
+          io-backends:
+            description:
+              Used to pipe external sigma delta modulator or internal ADC backend to MDF
+              channel.
+            maxItems: 1
+
+        required:
+          - reg
+
+        unevaluatedProperties: false
+
     allOf:
       - if:
           properties:
@@ -225,49 +254,11 @@ patternProperties:
                 const: st,stm32mp25-mdf-adc
 
         then:
-          properties:
-            '#address-cells':
-              const: 1
-
-            '#size-cells':
-              const: 0
-
-          required:
-            - '#address-cells'
-            - '#size-cells'
-
           patternProperties:
-            "^channel@([0-9])$":
-              type: object
-              $ref: "adc.yaml"
-              description: Represents the external channel which is connected to the MDF.
-
-              properties:
-                reg:
-                  maxItems: 1
-                  items:
-                    minimum: 0
-                    maximum: 8
-
-                settling-time-us:
-                  description:
-                    Time between enabling the channel and the first sample. This time is converted
-                    by the driver into a minimum number of output sample to discard before releasing
-                    the data. It allows to drop the samples affected by the impulse response of the
-                    filter and wait for stable data.
-
-                io-backends:
-                  description:
-                    Used to pipe external sigma delta modulator or internal ADC backend to MDF
-                    channel.
-                  maxItems: 1
-
+            "^channel@[0-7]$":
               required:
-                - reg
                 - io-backends
 
-              additionalProperties: false
-
       - if:
           properties:
             compatible:
@@ -340,6 +331,12 @@ examples:
         dma-names = "rx";
         st,cic-mode	= <5>;
         st,sitf = <&sitf5 0>;
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        channel@0 {
+          reg = <0>;
+        };
 
         asoc_pdm0: mdf-dai {
           compatible = "st,stm32mp25-mdf-dai";
